product-service/src/functions/createProduct/handler.ts

```diff
@@ -0,0 +1,86 @@
+import type { ValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
+import { formatJSONResponse } from '@libs/api-gateway';
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { DynamoDBDocumentClient, PutCommand } from '@aws-sdk/lib-dynamodb';
+import { v4 as uuidv4 } from 'uuid';
+import type { IProduct, IProductStock, IProductWithStock } from 'src/types/product';
+import { APIGatewayProxyResult } from 'aws-lambda';
+
+const client = new DynamoDBClient();
+const docClient = DynamoDBDocumentClient.from(client, {
+  marshallOptions: {
+    removeUndefinedValues: true,
+  },
+});
+
+const updateProductsTable = async (product: IProduct) => {
+  const command = new PutCommand({
+    TableName: process.env.PRODUCTS_TABLE,
+    Item: product,
+  });
+
+  await docClient.send(command);
+};
+
+const updateProductsStockTable = async (product: IProductStock) => {
+  const command = new PutCommand({
+    TableName: process.env.PRODUCTS_STOCK_TABLE,
+    Item: product,
+  });
+
+  await docClient.send(command);
+};
+
+const checkFieldAndRespond = (
+  fieldName: string,
+  payload: undefined | string | number,
+  callback: (error?: string | Error, result?: APIGatewayProxyResult) => void,
+) => {
+  if (payload === undefined) {
+    callback(null, {
+      statusCode: 400,
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ error: `Product data is invalid. ${fieldName} is missing` }),
+    });
+  }
+};
+
+const createProduct: ValidatedEventAPIGatewayProxyEvent<any> = async (
+  event,
+  _context,
+  callback,
+) => {
+  // @ts-ignore
+  const payload: IProductWithStock = JSON.parse(event.body);
+  checkFieldAndRespond('Description', payload.description, callback);
```

Comment:
there are different libs for handling if schema is correct, you can use some of them,
if not to avoid calling same function multiple time you can iterate over object to detect if property present, but you also need to check not only if prop exit, but type also